### PR TITLE
Do not force C++20 even if supported by compiler

### LINKED ISSUE #55: Including lexy in CMake project forces use of C++20
In `src/CMakeLists.txt:123`, the lexy CMake project checks whether the cxx compiler supports c++20 and if so, unconditionally enables it. This in turn gets propagated as part of the INTERFACE properties of the lexy targets, which means even if the user has specified C++17 for the toplevel project, this gets overriden by lexy.

The [CMake docs for 3.8](https://cmake.org/cmake/help/v3.8/manual/cmake-compile-features.7.html#requiring-language-standards) suggest using any of the cxx_std_?? compiler features guarantees at least that version will be used for compilation so it should be fine to just use cxx_std_17 in the compile features and let the user choose C++20 on her own. [More recent docs](https://cmake.org/cmake/help/v3.23/manual/cmake-compile-features.7.html#requiring-language-standards) make this more explicit.

### PR DESCRIPTION
**Pull request by lacc97**
Fixes #55.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -120,11 +120,7 @@ set(ext_header_files
 # Base target for common options.
 add_library(_lexy_base INTERFACE)
 target_sources(_lexy_base INTERFACE ${header_files})
-if ("cxx_std_20" IN_LIST CMAKE_CXX_COMPILE_FEATURES)
```
Comment (foonathan) [inline src/CMakeLists.txt:123]: Based on the discussion in #55, please make the proposed behavior conditional by introducing a CMake option `LEXY_FORCE_CPP17`.

**Review by foonathan — CHANGES_REQUESTED** [review]

File: src/CMakeLists.txt (hunk 1 above)
Comment (lacc97) [reply src/CMakeLists.txt:123]: To clarify, `LEXY_FORCE_CPP17=ON` means to use `target_compile_options(_lexy_base INTERFACE cxx_std_17)`, and in the `OFF` case, to use the old behaviour. Is this correct?

Comment (foonathan) [reply src/CMakeLists.txt:123]: Yes, and the option is off by default.

**Review by foonathan — APPROVED** [review]

**Comment by foonathan** [comment]
Thank you.

**Comment by zzk666666** [comment]
Sir,can you teach me C++,What style your write programs?I want study your programs style.Thx.

**Pull request merged**